lib/sprockets/npm.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+require 'json'
+
+module Sprockets
+  module Npm
+    # Internal: Override resolve_alternates to install package.json behavior.
+    #
+    # load_path    - String environment path
+    # logical_path - String path relative to base
+    #
+    # Returns candiate filenames.
+    def resolve_alternates(load_path, logical_path)
+      candidates, deps = super
+
+      # package.json can only be nested one level deep
+      if !logical_path.index('/'.freeze)
+        dirname = File.join(load_path, logical_path)
+
+        if directory?(dirname)
```

Comment:
I think this will also end up catching this case https://github.com/rails/sprockets/issues/169, maybe we could resolve that here as well?